### PR TITLE
Refactor molecule printing

### PR DESCRIPTION
**Pull request by xdvriend**
**Short description**
A small PR to address issue #610 

**Related issues**
#610 

**Don't forget:**
- [x] if new files are created: update the Doxygen file, the collective gqcp.hpp header and the CMake files

### CODE REVIEW

File: gqcp/include/Molecule/Molecule.hpp [hunk 1]
```diff
@@ -132,6 +134,19 @@ class Molecule {
...
     int charge() const;
 
+    /**
+     *  @return a textual description of this molecule.
+     */
+    std::string description() const {
+
+        std::string molecule_string = (boost::format("Number of electrons: %s \n") % this->numberOfElectrons()).str(); 
+        std::ostringstream framework;
+        framework << this->nuclearFramework();
```
Comment (lelemmen) [inline gqcp/include/Molecule/Molecule.hpp:144]: Could you also refactor this into `NuclearFramework::description()`?

**Review by lelemmen — COMMENTED** [review]
Great job! I see you've only refactored `Molecule::description()`. In the light of improving the code's durability, it's also a good idea to give a `description()` method to `NuclearFramework`.

File: gqcp/include/Molecule/Molecule.hpp [hunk 2]
```diff
@@ -20,6 +20,8 @@
 
 #include "Molecule/NuclearFramework.hpp"
 
+#include <boost/format.hpp>
```
Comment (dariatols) [inline gqcp/include/Molecule/Molecule.hpp:23]: I still don't understand why this line should be included. Is there a way that this code works without this `include`?

File: gqcp/src/Molecule/Molecule.cpp [hunk 3]
```diff
@@ -138,8 +138,7 @@ Molecule Molecule::ReadXYZ(const std::string& xyz_filename, const int charge) {
  */
 std::ostream& operator<<(std::ostream& os, const Molecule& molecule) {
 
-    os << "Number of electrons: " << molecule.numberOfElectrons() << std::endl;
-    os << molecule.nuclearFramework();
+    os << molecule.description();
```
Comment (dariatols) [inline gqcp/src/Molecule/Molecule.cpp:141]: Great job! This looks way cleaner now.

**Review by dariatols — APPROVED** [review]

File: gqcp/include/Molecule/Molecule.hpp (hunk 2 above)
Comment (xdvriend) [reply gqcp/include/Molecule/Molecule.hpp:23]: This include allows for line 142 in Molecule.hpp to be written like it is.

File: gqcp/include/Molecule/Molecule.hpp (hunk 1 above)
Comment (xdvriend) [reply gqcp/include/Molecule/Molecule.hpp:144]: Allright, will do!

**Review by lelemmen — APPROVED** [review]

**Pull request merged**